src/services/domain/DomainService.ts

```diff
@@ -38,6 +38,22 @@ export class DomainService {
     return deferred.promise;
   }
 
+  getHosts(): ng.IPromise<string []> {
+    let deferred: ng.IDeferred<string []> = this.$q.defer<string []>();
+    let request: IDmrRequest = <IDmrRequest>{
+      address: [],
+      "child-type": "host"
+    };
+    let hosts: string [] = [];
+    this.dmrService.readChildResources(request).then((response) => {
+      for (let hostname in response) {
+        hosts.push(hostname);
+      }
+      deferred.resolve(hosts);
+    });
```

Comment:
I think we can make use of `Object.keys` here and turn this into a one-liner, something like:

```typescript
this.dmrService.readChildResources(request).then(response => deferred.resolve(Object.keys(response)));
```